src/pollux/models/JWTVerifiableCredential.ts

```diff
@@ -4,159 +4,426 @@ import {
+        typeof payload[JWTVerifiableCredentialProperties.sub] !== 'string') {
+        throw new InvalidCredentialError("Invalid sub in credential payload should be string");
+      }
+
+
+      if (typeof payload[JWTVerifiableCredentialProperties.aud] !== 'undefined' &&
+        (typeof payload[JWTVerifiableCredentialProperties.aud] !== 'string' &&
+          !Array.isArray(payload[JWTVerifiableCredentialProperties.aud]))) {
+        throw new InvalidCredentialError("Invalid aud in credential payload should be string");
+      }
+
+
+      if (typeof payload[JWTVerifiableCredentialProperties.revoked] !== 'undefined' &&
+        typeof payload[JWTVerifiableCredentialProperties.revoked] !== 'boolean') {
+        throw new InvalidCredentialError("Invalid revoked in credential payload should be boolean");
+      }
+
+      //TODO: Improve validation of VC
+      if (typeof payload[JWTVerifiableCredentialProperties.vc] !== 'undefined' &&
+        typeof payload[JWTVerifiableCredentialProperties.vc] !== 'object') {
+        throw new InvalidCredentialError("Invalid vc in credential payload should be an object");
+      }
+
+    } else {
+      if (typeof payload[JWTVerifiablePresentationProperties.iss] !== 'undefined' &&
+        typeof payload[JWTVerifiablePresentationProperties.iss] !== 'string') {
+        throw new InvalidCredentialError("Invalid iss in presentation payload should be string");
+      }
+
+      if (typeof payload[JWTVerifiablePresentationProperties.aud] !== 'undefined' &&
+        (typeof payload[JWTVerifiablePresentationProperties.aud] !== 'string' &&
+          !Array.isArray(payload[JWTVerifiablePresentationProperties.aud]))) {
+        throw new InvalidCredentialError("Invalid aud in presentation payload should be string");
+      }
+
+      if (typeof payload[JWTVerifiablePresentationProperties.nonce] !== 'undefined' &&
+        typeof payload[JWTVerifiablePresentationProperties.nonce] !== 'string') {
+        throw new InvalidCredentialError("Invalid nonce in presentation payload should be string");
+      }
+
+      if (typeof payload[JWTVerifiablePresentationProperties.nbf] !== 'undefined' &&
+        typeof payload[JWTVerifiablePresentationProperties.nbf] !== 'number') {
+        throw new InvalidCredentialError("Invalid nbf in presentation payload should be number");
+      }
+
+      if (typeof payload[JWTVerifiablePresentationProperties.exp] !== 'undefined' &&
+        typeof payload[JWTVerifiablePresentationProperties.exp] !== 'number') {
+        throw new InvalidCredentialError("Invalid exp in presentation payload should be number");
+      }
+
+      //TODO: Improve validation of VP
+      if (typeof payload[JWTVerifiablePresentationProperties.vp] !== 'undefined' &&
+        typeof payload[JWTVerifiablePresentationProperties.vp] !== 'object') {
+        throw new InvalidCredentialError("Invalid vp in presentation payload should be an object");
+      }
```

Comment:
wouldn't it be cleaner to create a JWTVerifiablePresentation class, instead of adding 100's of conditionals?

Reply:
Well, we can create another class but this class at the end will not be typed in Javascript and must be validated accordingly, that would just move the logic somewhere else, introduce another resource we'd need to store in the database and is too much change, don't think its worth honestly :)

Reply:
JWTCredential class is not instantiated either with a string (JWS) a presentation or a credential payload, but again for more than we write there that the input type is JWTPresentationPayload this means nothing in JS at runtime